### PR TITLE
Ensure high level triggers project their user state

### PR DESCRIPTION
**Pull request by carlpulley-da**
Currently, trigger logging prints the raw `SValue` state. For high level triggers this means that the ACS will be logged! This PR ensures the user state is projected from high level trigger state.

This PR has been cherry picked from https://github.com/digital-asset/daml/pull/15564

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ import com.daml.lf.data.Ref
 import com.daml.lf.data.Ref._
 import com.daml.lf.data.ScalazEqual._
 import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.engine.trigger.Runner.triggerUserState
```
Comment (akrmn) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:32]: I'm a bit surprised that you have to (or even that you can) import `triggerUserState` from _this very file_, why is that?

**Review by akrmn — APPROVED** [review]
LGTM, just have a scala question probably showing my ignorance. Thanks Carl!

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala (hunk 1 above)
Comment (carlpulley-da) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:32]: `triggerUserState` should be private (as we have no intention of anyone else using it) - so I'll push out that small change.

That said, `triggerUserState` is defined in the `Runner` classes companion object. To access this from an instance of the `Runner` class, we need to either import it (as we do here) or use the fully qualified path (i.e. `Runner.triggerUserState`). Does that answer your question?

Comment (akrmn) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala:32]: aha, I'd missed that this file defines both `Runner`-the-class and `Runner`-the-object, thanks for explaining!

**Pull request merged**